crates/neon/src/executor/mod.rs

```diff
@@ -0,0 +1,59 @@
+use std::{future::Future, pin::Pin};
+
+use crate::{context::Cx, thread::LocalKey};
+
+#[cfg(feature = "tokio-rt")]
+pub(crate) mod tokio;
+
+type BoxFuture = Pin<Box<dyn Future<Output = ()> + Send + 'static>>;
```

Comment:
I need the `Runtime` trait to be object-safe so that I can put it in a `Box<dyn Runtime>`; this means that the `spawn(..)` function can't be generic. Since it can't be generic, it takes a boxed dyn future.

Boxed futures need be `Pin` because of [`Future::poll`](https://doc.rust-lang.org/stable/std/future/trait.Future.html#tymethod.poll) which takes `self: Pin<&mut Self>`. Without the `Pin`, it could be moved in memory and you wouldn't be able to poll it.

This is a very common type in async Rust. The `futures` crate even includes a [type alias for it](https://docs.rs/futures-core/0.3.30/futures_core/future/type.BoxFuture.html).